lib/utils/HDFDatabaseMPIO.h

```diff
@@ -0,0 +1,455 @@
+/******************************************************************************
+ *
+ * Copyright (c) 2013-2024, Lawrence Livermore National Security, LLC
+ * and other libROM project developers. See the top-level COPYRIGHT
+ * file for details.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR MIT)
+ *
+ *****************************************************************************/
+
+// Description: The concrete database implementation using HDF5.
+
+#ifndef included_HDFDatabaseMPI_h
+#define included_HDFDatabaseMPI_h
+
+#include "HDFDatabase.h"
+
+namespace CAROM {
+
+/**
+ * HDFDatabaseMPIO implements Multi-Path Input/Output (MPIO) of HDF5.
+ */
+class HDFDatabaseMPIO : public HDFDatabase
```

Comment:
I noticed compiling this branch with the serial version of HDF5 fails. I saw you updated the Dockerfile dependencies to install the parallel version which is good, but should we support compiling with the serial version of HDF5 still? If so, we can modify this a bit to conditionally include the MPIO related things depending on whether the H5 library has parallel support using `H5_HAVE_PARALLEL`. It might require several ifdef statements around the code and in BasisReader/BasisWriter though.
If we don't want to do that now, then maybe we can make sure that parallel HDF5 is found by CMake. I think adding something like this to the main `CMakeLists.txt` after the find package statement for HDF5 should work:
```
if(NOT ${HDF5_IS_PARALLEL})
  message(FATAL_ERROR "HDF5 library does not support MPI - please use parallel HDF5")
endif()
```